apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -0,0 +1,367 @@
+
+        /// <summary>
+        /// </summary>
+        public new SpannerParameterCollection Parameters { get; private set; }
+
+        /// <summary>
+        /// </summary>
+        public SpannerConnection SpannerConnection { get; set; }
+
+        /// <inheritdoc />
+        public override UpdateRowSource UpdatedRowSource
+        {
+            get { throw new NotImplementedException(); }
+            set { throw new NotImplementedException(); }
+        }
+
+        /// <inheritdoc />
+        protected override DbConnection DbConnection
+        {
+            get { return SpannerConnection; }
+            set { SpannerConnection = (SpannerConnection) value; }
+        }
+
+        /// <inheritdoc />
+        protected override DbParameterCollection DbParameterCollection => Parameters;
+
+        /// <inheritdoc />
+        protected override DbTransaction DbTransaction
+        {
+            get { return _transaction; }
+            set { _transaction = (SpannerTransaction) value; }
+        }
+
+        /// <summary>
+        /// </summary>
+        internal SpannerCommandTextBuilder SpannerCommandTextBuilder { get; set; }
+
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        public object Clone()
+        {
+            return new SpannerCommand {
+                DesignTimeVisible = DesignTimeVisible,
+                Parameters = Parameters,
+                SpannerConnection = SpannerConnection,
+                SpannerCommandTextBuilder = SpannerCommandTextBuilder,
+                CommandTimeout = CommandTimeout
+            };
+        }
+
+        /// <inheritdoc />
+        public override void Cancel()
+        {
+            _synchronousCancellationTokenSource.Cancel();
+        }
+
+        /// <inheritdoc />
+        public override int ExecuteNonQuery()
+        {
+            return ExecuteNonQueryAsync(_synchronousCancellationTokenSource.Token).Result;
+        }
+
+        /// <inheritdoc />
+        public override async Task<int> ExecuteNonQueryAsync(CancellationToken cancellationToken)
+        {
+            // There must be a valid and open connection.
+            if (SpannerConnection == null)
+                throw new InvalidOperationException(
+                    "You must assign a SpannerConnection to this command to execute it.");
+            if (SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Delete
+                && SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Insert
+                && SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.InsertOrUpdate
+                && SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Update)
+                throw new InvalidOperationException(
+                    "You can only call ExecuteReader on a Delete, Insert, InsertUpdate, or Update Command");
+            if (!SpannerConnection.IsOpen)
```

Comment:
right, my current VS/resharper settings are actually removing them when possible.  If we do not have them already, lets create a team standard for these settings files.